infrastructure-scripts/New-SqlDbServiceAccount.ps1

```diff
@@ -79,79 +60,85 @@ function Update-AzureSQLServerFirewallRule {
     $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
     Set-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
 }
-
 function Get-RandomPassword {
     $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
     return $Password
 }
-
 function Remove-AzureSQLServerFirewallRule {
     if ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue)) {
         Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
     }
 }
 
 try {
-    $SecretName = "${Enviroment}-${SqlServiceAccountName}"
 
-    Write-Host "$SecretName"
+    $ServiceAccountSecretName = "$Enviroment-$SqlServiceAccountName".ToLower()
+    $ServerFQDN = "$ServerName.database.windows.net"
 
-    if ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue) -eq $null) {
+    # --- Retrieve SQL Server details
+    Write-Host "Searching for server resource $($ServerName)"
+    $ServerResource = Get-AzureRmResource -Name $ServerName -ResourceType "Microsoft.Sql/servers"
```

Comment:
Look at README guidelines for casing rules on local variables

Reply:
Don't fully agree with that - we should discuss @dalemorson @EwanNoble

Reply:
Ref #7